src/compiler/checker.ts

```diff
@@ -4583,7 +4583,22 @@ namespace ts {
                 let requireOptionalProperties = relation === subtypeRelation && !(source.flags & TypeFlags.ObjectLiteral);
                 for (let targetProp of properties) {
                     let sourceProp = getPropertyOfType(source, targetProp.name);
-                    if (sourceProp !== targetProp) {
+
+                    if (sourceProp === targetProp) { // source inherits targetProp and doesn't redeclare/override it.

```

Comment:
Why do we need to check this. there are no instances of the abstract class created. so the fact that the type has a property whose declaration is abstract should not affect assignability.